examples/graph/plot_DAG_layouts.py

```diff
@@ -0,0 +1,63 @@
+"""
+======================
+DAG in Various Layouts
+======================
+Creates a Directed Acyclic Graph and shows the Graph in various layouts. 
+This can give an idea as to which layout can be chosen for these types of datasets.
+"""
+
+import networkx as nx
+import matplotlib.pyplot as plt
+
+
+G = nx.DiGraph()
+G.add_edges_from(
+    [
+        ("f", "a"),
+        ("a", "b"),
+        ("a", "e"),
+        ("b", "c"),
+        ("b", "d"),
+        ("d", "e"),
+        ("f", "c"),
+        ("f", "g"),
+        ("h", "f"),
+    ]
+)
+
+
+fig, axes = plt.subplots(2, 4, figsize=(25, 12))
+layouts = [
+    nx.spring_layout(G),
+    nx.circular_layout(G),
+    nx.kamada_kawai_layout(G),
+    nx.planar_layout(G),
+    nx.random_layout(G),
+    nx.shell_layout(G),
+    nx.spectral_layout(G),
+    nx.spiral_layout(G),
+]
+titles = [
+    "Spring Layout - Default when given pos = None",
+    "Circular Layout",
+    "Kamada Kawai Layout",
+    "Planar Layout",
+    "Random Layout",
+    "Shell Layout",
+    "Spectral Layout",
+    "Spiral Layout",
+]
```

Comment:
Another (very minor) comment here - I think this would be improved by using a dict here instead of two lists. The motivation for doing so is that it is, in principle at least, easier to guarantee that the titles and the layouts remain in sync. Instead of having to manually guarantee the ordering is correct in two lists, you can just make sure that the key: value pairs are correct in the dict.

This would require reworking the unpacking in the for-loop below, but that should be doable!

Also, another unrelated comment: instead of storing the pos-dicts themselves, it would be slightly more memory-efficient to store the layout *functions* here, then call them in the for-loop.

Reply:
Hi, I didn't quite catch the last sentence. It is said to store the layout function here ... here as in? I have converted the code to include a dict and changed for loop accordingly. I didn't quite get how I can improve the memory efficiency. 

```suggestion
lay = {
    "Spring Layout - Default when given pos = None": nx.spring_layout(G),
    "Circular Layout": nx.circular_layout(G),
    "Kamada Kawai Layout": nx.kamada_kawai_layout(G),
    "Planar Layout": nx.planar_layout(G),
    "Random Layout": nx.random_layout(G),
    "Shell Layout": nx.shell_layout(G),
    "Spectral Layout": nx.spectral_layout(G),
    "Spiral Layout": nx.spiral_layout(G),
}

```

Reply:
I think the suggestion is to avoid computing all of these layouts until they are needed.  By making the value of the dict be a function, that lets you use the function later to find the layout.

The advantage is that if you don't use that layout, you don't waste a lot of time computing the layout when you create the dict.  In this example, you do actually use all the layouts. But it is a nice style to avoid computing something until it is needed. And having a dict keyed by name to a function is a common way to provide options when all the options have the same function signature.

Reply:
Exactly, thanks @dschult 

More concretely what we mean is - instead of including the results of the layouts as values in the dictionary, use the layout *functions* as the values, then call the function in the for-loop, e.g.:

```python
layouts = {
    "Spring layout": nx.spring_layout,
    # ... etc.
}
for (title, layout_fn), ax in zip(layouts.items(), ax.flatten()):  # Or something like this
    pos = layout_fn(G)
    # ...
```

Reply:
Kkay thank you! I believe I understand now. I have made the necessary changes in file :)